improver_tests/threshold/test_ThresholdPytest.py

```diff
@@ -0,0 +1,599 @@
+        ({"threshold_config": {"0.6": [0.4, 0.7]}}, 0.25, [0.125, 0.375]),
+        # asymmetric fuzzy bounds applied, diagnostic value(s) above threshold value
+        ({"threshold_config": {"0.4": [0.0, 0.6]}}, 0.75, [0.625, 0.875]),
+        # asymmetric fuzzy bounds applied, diagnostic value(s) at threshold value
+        ({"threshold_config": {"0.5": [0.4, 0.7]}}, 0.5, [0.25, 0.625]),
+    ],
+)
+def test_expected_values(default_cube, kwargs, collapse, comparator, expected_result):
+    """Test that thresholding yields the expected data values for
+    different configurations. Variations tried here are:
+
+      - Different threshold values relative to the diagnostic value(s)
+      - Use of fuzzy thresholds, specified in different ways.
+      - Deterministic, single realization, and multi-realization inputs
+      - Different threshold comparators. Note that the tests have been
+        engineered such that there are no cases where the difference
+        between "ge" and "gt", or "le" and "lt" are signficant, these
+        are tested elsewhere.
+      - Collapsing and not collapsing the realization coordinate when
+        present.
+    """
+
+    local_kwargs = kwargs.copy()
+
+    if collapse and default_cube.coords("realization", dim_coords=True):
+        local_kwargs.update({"collapse_coord": "realization"})
+    elif collapse:
+        # No need to repeat tests in which collapse_coord is not used.
+        pytest.skip()
+
+    local_kwargs.update({"comparison_operator": comparator})
+    plugin = Threshold(**local_kwargs)
+    result = plugin(default_cube)
+
+    assert result.data.shape == expected_result.shape
+    assert np.allclose(result.data, expected_result)
+
+
+@pytest.mark.parametrize(
+    "kwargs,n_realizations,data,expected_result",
+    [
+        # diagnostic value at threshold value, no fuzziness, various comparators
+        (
+            {"threshold_values": 0.0, "comparison_operator": "gt"},
+            1,
+            np.zeros((3, 3)),
+            np.zeros((3, 3), dtype=np.float32),
+        ),
+        (
+            {"threshold_values": 0.0, "comparison_operator": "lt"},
+            1,
+            np.zeros((3, 3)),
+            np.zeros((3, 3), dtype=np.float32),
+        ),
+        (
+            {"threshold_values": 0.0, "comparison_operator": "ge"},
+            1,
+            np.zeros((3, 3)),
+            np.ones((3, 3), dtype=np.float32),
+        ),
+        (
+            {"threshold_values": 0.0, "comparison_operator": "le"},
+            1,
+            np.zeros((3, 3)),
+            np.ones((3, 3), dtype=np.float32),
+        ),
+        # diagnostic value at threshold value of 0, with fuzziness bounding.
+        # All comparators will give the same result for this symmetric case.
+        (
+            {"threshold_config": {"0.": [-1, 1]}, "comparison_operator": "gt"},
+            1,
+            np.zeros((3, 3)),
+            np.ones((3, 3), dtype=np.float32) * 0.5,
```

Comment:
Recommended approach for all your `ones * value`:
```suggestion
            np.full((3, 3), 0.5, dtype=np.float32),
```